src/main/java/com/prueba/springboot2/dao/IClientDAO.java

```diff
@@ -0,0 +1,13 @@
+package com.prueba.springboot2.dao;
+
+import java.util.List;
+import com.prueba.springboot2.entity.Client;
+
+public interface IClientDAO {
```

Comment:
En vez de crear un DAO deberias usar spring data y hacer que esta interfaz extienda de otra 'JpaRepository<T, ID>'
donde T es el tipo de la clase de modelo para la cual queres crear tu repository e ID es el tipo del idenfitiador o 'id'  de esa misma clase, example:

````
public interfacte ClientRepository extends JpaRepository<Client, Integer> {

}
````

Una vez hagas esto spring data de va a proveer funcionalidades out-of-the-box como, findOne, findAll, deleteOne, deleteAll, etc.